rules/integrations/google_workspace/initial_access_google_workspace_object_copied_from_external_drive_and_access_granted_to_custom_application.toml

```
@@ -0,0 +1,129 @@
+been added to also detect when a user grants a custom Google application permission via OAuth shortly after. An
+adversary may send a phishing email to the victim with a Drive object link where "copy" is included in the URI, thus
+copying the object to the victim's drive. If a container-bound script exists within the object, execution will require
+permission access via OAuth in which the user has to accept.
+"""
+false_positives = [
+    """
+    Google Workspace users typically share Drive resources with a shareable link where parameters are edited to indicate
+    when it is viewable or editable by the intended recipient. It is uncommon for a user in an organization to manually
+    copy a Drive object from an external drive to their corporate drive. This may happen where users find a useful
+    spreadsheet in a public drive, for example, and replicate it to their Drive.
+    """,
+]
+from = "now-130m"
+index = ["filebeat-*", "logs-google_workspace*"]
+interval = "10m"
+language = "eql"
+license = "Elastic License v2"
+name = "Google Workspace Object Copied from External Drive and Access Granted to Custom Application"
+note = """## Triage and analysis
+
+### Investigating Google Workspace Resource Copied from External Drive and Access Granted to Custom Application
+
+Google Workspace user's have the capability to share access to Drive objects such as documents, sheets, forms and more. Delivery of shared access can be done by email delivery or via shared link. Shared link URIs have `view` or `edit` in the parameters based on set permissions. The `copy` parameter is another option which allows the recipient to copy the object to their own Drive, thus the object has the privileges of the recipient. Specific objects allow container-bound scripts that leverage Google's Apps Script platform. Container-bound scripts can contain malicious code that executes with the recipient's privileges if in their Drive.
+
+This rule detections when a user copies an external Drive object to their Drive storage, followed by granting permissions to a custom application via OAuth prompt.
+
+#### Possible investigation steps
+- Identify user account(s) associated by reviewing `user.name` or `source.user.email` in the alert.
+- Identify the name of the file copied by reviewing `file.name` as well as the `file.id` for triaging.
+- Identify the file type by reviewing `google_workspace.drive.file.type`.
+- With the information gathered so far, query across data for the file metadata to determine if this activity is isolated or widespread.
+- Within the OAuth token event, identify the application name by reviewing `google_workspace.token.app_name`.
+    - Review the application ID as well from `google_workspace.token.client.id`.
+    - This metadata can be used to report the malicious application to Google for permanent blacklisting.
+- Review what permissions were granted access by the user by reviewing `google_workspace.token.scope.data.scope_name`.
```

```suggestion
- Identify the permissions granted to the application by the user by reviewing `google_workspace.token.scope.data.scope_name`.
```
suggested change for clarity and uniformity